lib/rules/id-length.js

```diff
@@ -61,7 +61,7 @@ module.exports = {
                 return !parent.computed && (
 
                     // regular property assignment
-                    (parent.parent.left === parent || // or the last identifier in an ObjectPattern destructuring
+                    (parent.parent.left === parent && parent.parent.type === "AssignmentExpression" ||  // or the last identifier in an ObjectPattern destructuring
```

Comment:
I know this isn't something you changed, but the location of this comment is weird - maybe add it to the comment above?